File: pong/servidor.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
	"sync"
	"time"
)

type LogEntry struct {
	Time     string `json:"time"`
	Method   string `json:"method"`
	Path     string `json:"path"`
	Response string `json:"response"`
}

var (
	logs []LogEntry
	mu   sync.Mutex
)

func healthHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Access-Control-Allow-Origin", "*")
	entry := LogEntry{
		Time:     time.Now().Format("2006-01-02 15:04:05"),
		Method:   r.Method,
		Path:     r.URL.Path,
		Response: "Healthcheck OK",
	}

	mu.Lock()
	logs = append(logs, entry)
	mu.Unlock()

	fmt.Printf("[%s] %s %s -> %s\n",
		entry.Time,
		entry.Method,
		entry.Path,
		entry.Response,
	)

	w.Header().Set("Content-Type", "text/plain")
	w.Write([]byte("PONG"))
}

func logsHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Access-Control-Allow-Origin", "*")
	w.Header().Set("Content-Type", "application/json")
	mu.Lock()
	defer mu.Unlock()
	json.NewEncoder(w).Encode(logs)
}

func main() {

	http.HandleFunc("/health", healthHandler)
	http.HandleFunc("/logs", logsHandler)

	fs := http.FileServer(http.Dir("./src/frontend"))
	http.Handle("/", fs)

	fmt.Println("Servidor arriba en http://localhost")

	err := http.ListenAndServe(":8080", nil)
	if err != nil {
		panic(err)
	}
}

File: ping/src/servidor.py
from flask import Flask, jsonify, send_from_directory, request
from datetime import datetime
import threading
import os

app = Flask(__name__, static_folder="./frontend", static_url_path="")

class LogEntry:
    def __init__(self, method, path, response):
        self.time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        self.method = method
        self.path = path
        self.response = response

    def to_dict(self):
        return {
            "time": self.time,
            "method": self.method,
            "path": self.path,
            "response": self.response
        }

logs = []
lock = threading.Lock()

@app.route("/health", methods=["GET"])
def health():
    entry = LogEntry(
        method=request.method,
        path=request.path,
        response="Healthcheck OK"
    )

    with lock:
        logs.append(entry.to_dict())

    print(
        f"[{entry.time}] "
        f"{entry.method} "
        f"{entry.path} -> "
        f"{entry.response}"
    )

    return "PONG", 200

@app.route("/logs", methods=["GET"])
def get_logs():
    with lock:
        return jsonify(logs)

@app.route("/")
def serve_index():
    return send_from_directory("./frontend", "index.html")

@app.route("/<path:path>")
def serve_static(path):
    return send_from_directory("./frontend", path)

if __name__ == "__main__":
    print("Servidor arriba en http://localhost")

    app.run(
        host="0.0.0.0",
        port=80
    )